### PR TITLE
Implemented distribution_plot() and count_plot()

### PR DESCRIPTION
**Pull request by Mfkessler**
First version of plotting functions, written during Hackathon 09.2022

### CODE REVIEW

**Review by hschult — CHANGES_REQUESTED** [review]

**Comment by hschult** [comment]
Overall please add some comments to your code. It is a good habit to add a comment every few lines. Always try to write your code in a way that you can easily come back in a few months and understand what you did.

File: uropa/visualization.py [hunk 1]
```diff
@@ -3,52 +3,102 @@
...
-    <datatype> :
-        <return description>
-    TODO should return the plotting object
+    matplotlib.pyplot :
+        pyplot object for further processing
     """
-    pass
+    sns.set(style = "darkgrid")
+    
+    match kind:
```
[the remark this answers is not in the document]
Comment (hschult) [reply uropa/visualization.py:34]: Please add this version restriction to setup.py. Also, add your dependencies to `install_requires`.

**Review by hschult — CHANGES_REQUESTED** [review]

File: uropa/visualization.py [hunk 2]
```diff
@@ -1,54 +1,106 @@
...
+        case "histogram":
+                distPlot = sns.histplot(data=table[var])
+        case "boxplot":
+                distPlot = sns.boxplot(y=table[var])
+        case "violin":
+                distPlot = sns.violinplot(x=table[var])
+        case _:
+            print(f"{kind} not supported. Consider using one of the supported plots (histogram, boxplot or violin).")
```
Comment (hschult) [inline uropa/visualization.py:42]: Please, raise an error. Don't use `exit` for this.

**Review by hschult — CHANGES_REQUESTED** [review]

File: uropa/visualization.py [hunk 3]
```diff
@@ -3,52 +3,102 @@
...
     Returns
     -------
-    <datatype> :
-        <return description>
-    TODO should return the plotting object
+    matplotlib.pyplot :
+        pyplot object for further processing
     """
-    pass
+    sns.set(style = "darkgrid")
```
[the remark this answers is not in the document]
Comment (Mfkessler) [reply uropa/visualization.py]: actually it does not seem to be default using jupyter notebook, so I added it again.

File: uropa/visualization.py [hunk 4]
```diff
@@ -1,77 +1,125 @@
...
-    """
-    pass
+    if output:
+        plt.savefig(output, dpi=dpi)
+
+    return distPlot
 
 
-def peak_count_plot(table, var, kind):
+def peak_count_plot(table, var="feature", peak_type="exon", group_by=["peak_chr", "peak_start", "peak_end", "peak_strand"], stacked=False, color_by=None, title=None, output=None, dpi=300):
```
Comment (hschult) [inline uropa/visualization.py:56]: check parameters `var, feature, peak_type, group_by, color_by`. Also, I can't select `nan` with `peak_type` as `nan` can only be checked with `.isna()`.

File: uropa/visualization.py [hunk 5]
```diff
@@ -1,77 +1,125 @@
...
+
+    # Filter DataFrame by column var and peak_type
+    df = table[table[var] == peak_type]
+
+    if color_by:
+        group_by.append(color_by)
+        # Count the number of equal columns which belong to the group_by list and add the column to the DataFrame
+        df = df.groupby(group_by).size().to_frame(peak_type)
+        # Count the corresponding numbers of the color_by values and add the column to the DataFrame ("count")
+        df = df.groupby([peak_type, color_by]).size().to_frame(
```
Comment (hschult) [inline uropa/visualization.py:97]: Why did you split the line? If it was due to length consider this:
https://stackoverflow.com/a/33944618/19870975

File: uropa/visualization.py [hunk 6]
```diff
@@ -1,77 +1,125 @@
...
+            "count").reset_index()
+
+        # Generate stacked barplot
+        if stacked:
+            pcPlot = df.pivot(index=peak_type, columns=color_by, values="count").plot(
```
Comment (hschult) [inline uropa/visualization.py:102]: See comment above

File: uropa/visualization.py [hunk 7]
```diff
@@ -1,77 +1,125 @@
...
+        df = df.groupby(group_by).size()
+        # Generate histogram
+        pcPlot = sns.histplot(data=df, discrete=True)
+        pcPlot.set(xlabel=peak_type, ylabel="count")
+
+    if title:
+        pcPlot.set(title=title)
+
+    if output:
+        plt.savefig(output, dpi=dpi)
```
Comment (hschult) [inline uropa/visualization.py:120]: `dpi` has only effect on the saved image. Please change that so it will always apply.

**Review by hschult — CHANGES_REQUESTED** [review]

**Review by hschult — APPROVED** [review]
Looks good to me. We can merge now.
Good job 🥳

**Pull request merged**